src/flint/fmpq_mpoly.jl

```diff
@@ -1059,3 +1078,47 @@ function (R::QQMPolyRing)(a::Vector{Any}, b::Vector{Vector{T}}) where T
 
   return R(newaa, newbb)
 end
+
+###############################################################################
+#
+#   Changing base ring, mapping coefficients
+#
+###############################################################################
+
+function map_coefficients(R::ZZRing, f::QQMPolyRingElem;
+                          cached::Bool = true,
+                          parent::ZZMPolyRing = AbstractAlgebra._change_mpoly_ring(R, parent(f), cached))
+  @req isinteger(_content_ptr(f)) "input polynomial must have integral coefficients"
+  @req ngens(parent) == ngens(Nemo.parent(f)) "parents must have matching numbers of generators"
+  @req internal_ordering(parent) == internal_ordering(Nemo.parent(f)) "parents must have matching internal ordering"
+  return mul!(zero(parent), f, 1)
+end
+
+function map_coefficients(F::fpField, f::QQMPolyRingElem;
+                          cached::Bool = true,
+                          parent::MPolyRing = AbstractAlgebra._change_mpoly_ring(F, parent(f), cached))
+  dF = denominator(f)
+  d = F(dF)
+  if iszero(d)
+    error("Denominator divisible by p!")
+  end
+  m = inv(d)
+  ctx = MPolyBuildCtx(parent)
+  for x in zip(coefficients(f), exponent_vectors(f))
+    el = numerator(x[1] * dF)
```

Comment:
Could optimize this further by using `_zpoly_ptr` (to avoid converting the internal `fmpz` coefficients into `fmpq`, and then back via a multiplication)